language/compiler/bytecode_source_map/src/disassembler.rs

```diff
@@ -111,17 +259,198 @@ impl<Location: Clone + Eq + Default> Disassembler<Location> {
                 .get(ty_param_index as usize)
                 .ok_or_else(|| {
                     format_err!(
-                        "Type parameter index out of bounds while disassembling type signature"
+                        "Type parameter index {} out of bounds while disassembling type signature",
+                        ty_param_index
                     )
                 })?
                 .0
                 .to_string(),
         })
     }
 
-    //***************************************************************************
-    // Disassemblers
-    //***************************************************************************
+    fn disassemble_instruction(
+        &self,
+        instruction: &Bytecode,
+        locals_sigs: &LocalsSignature,
+        function_source_map: &FunctionSourceMap<Location>,
+    ) -> Result<String> {
+        match instruction {
+            Bytecode::LdAddr(address_idx) => {
+                let address = self.source_mapper.bytecode.address_at(*address_idx);
+                Ok(format!("LdAddr({})", address))
+            }
+            Bytecode::LdByteArray(byte_array_idx) => {
+                let bytearray = self.source_mapper.bytecode.byte_array_at(*byte_array_idx);
+                Ok(format!("LdByteArray({:?})", bytearray))
+            }
+            Bytecode::LdStr(string_idx) => {
+                let string = self.source_mapper.bytecode.user_string_at(*string_idx);
+                Ok(format!("LdStr({})", string.as_str()))
+            }
+            Bytecode::CopyLoc(local_idx) => {
```

Comment:
In the (very helpful!) output in your summary, printing the type info for every use of a local seems like a bit much. We already have the type info either in the function sig or the locals declaration; might be ok to just use the names to keep the output small?

Reply:
Fair point. I think this is probably a good idea. What do you think @dariorussi ?